File: codeforces/ungrouped/cf2037b.cpp
#include <iostream>
#include <vector>
#include <map>
#include <cmath>
using namespace std;

int main() {
    int ntc;
    cin >> ntc;
    
    while (ntc--) {
        int k;
        cin >> k;
        

        vector<int> a;
        int temp;
        for (int i = 0; i < k; i++) {
            cin >> temp;
            a.push_back(temp);
        }
        

        map<int, int> count;
        for (int x : a) {
            count[x]++;
        }
        
        int area = k - 2;
        bool found = false;
        

        for (int i = 1; i <= sqrt(area); i++) {
            if (area % i == 0) {
                int j = area / i;
                if (count.find(i) != count.end() && count.find(j) != count.end()) {
                    if (i == j) {
                        if (count[i] >= 2) {
                            cout << i << " " << j << endl;
                            found = true;
                            break;
                        }
                    } else {
                        cout << i << " " << j << endl;
                        found = true;
                        break;
                    }
                }
            }
        }
    }
    return 0;
}

File: codeforces/ungrouped/cf2037b.py
from math import isqrt
from collections import Counter

ntc = int(input())
for _ in range(ntc):
    k = int(input())
    a = list(map(int, input().split()))
    count = Counter(a)
    area = k - 2

    for i in range(1, isqrt(area) + 1):
        if area % i == 0:
            j = area // i
            if i in count and j in count:
                if i == j:
                    if count[i] >= 2:
                        print(i, j)
                        break
                else:
                    print(i, j)
                    break
    else:
        print(-1, -1)


